source/4_tools/regina/src/regina.cpp

```diff
@@ -1,54 +1,865 @@
-  }
-  void shutdown()
-  {
-    REX_LOG(LogRegina, "shutting down Regina");
-  }
-
-  ApplicationCreationParams create_regina_app_creation_params(PlatformCreationParams&& platformParams)
-  {
-    ApplicationCreationParams app_params(rsl::move(platformParams));
-
-    app_params.gui_params.window_width  = 1280;
-    app_params.gui_params.window_height = 720;
-    app_params.gui_params.window_title  = "Regina";
-
-    app_params.engine_params.max_memory        = 256_kb;
-    app_params.engine_params.app_init_func     = initialize;
-    app_params.engine_params.app_update_func   = update;
-    app_params.engine_params.app_shutdown_func = shutdown;
-
-    app_params.create_window = true;
-
-    return app_params;
-  }
+    struct ObjectConstants
+    {
+        glm::mat4 world = glm::mat4(1.0f);
+    };
 
-#ifndef REX_ENABLE_AUTO_TESTS
+    struct PassConstants
+    {
+        glm::mat4 view = glm::mat4(1.0f);
+        glm::mat4 inv_view = glm::mat4(1.0f);
+        glm::mat4 proj = glm::mat4(1.0f);
+        glm::mat4 inv_proj = glm::mat4(1.0f);
+        glm::mat4 view_proj = glm::mat4(1.0f);
+        glm::mat4 inv_view_proj = glm::mat4(1.0f);
+
+        glm::vec3 eye_pos_w = { 0.0f, 0.0f, 0.0f };
+        f32 cb_padding_1 = 0.0f;
+
+        glm::vec2 render_target_size = { 0.0f, 0.0f };
+        glm::vec2 inv_render_target_size = { 0.0f, 0.0f };
+
+        f32 near_z = 0.0f;
+        f32 far_z = 0.0f;
+
+        f32 total_time = 0.0f;
+        f32 delta_time = 0.0f;
+    };
+
+    struct FrameData
+    {
+        renderer::ResourceSlot frame;
+        renderer::ResourceSlot object_commited_resource;
+        renderer::ResourceSlot pass_commited_resource;
+
+        rsl::vector<renderer::ResourceSlot> object_constant_buffer;
+        renderer::ResourceSlot pass_constant_buffer;
+    };
+
+    struct ReginaContext
+    {
+        rsl::vector<FrameData> frame_resource_data;
+
+        rsl::unordered_map<rsl::medium_stack_string, rsl::unique_ptr<renderer::Mesh>> meshes;
+
+        rsl::unique_ptr<renderer::Mesh> mesh_cube;
+        rsl::unique_ptr<renderer::Scene> scene;
+        rsl::unique_ptr<renderer::SceneRenderer> scene_renderer;
+
+        renderer::ResourceSlot clear_state;
+        renderer::ResourceSlot shader_program;
+        renderer::ResourceSlot input_layout;
+        renderer::ResourceSlot pso;
+        renderer::ResourceSlot solid_raster_state;
+        renderer::ResourceSlot wire_raster_state;
+
+        PassConstants pass_constants;
+
+        bool is_wireframe = false;
+
+        glm::vec3 eye_pos = { 0.0f, 0.0f, 0.0f };
+        glm::mat4 view = glm::mat4(1.0f);
+        glm::mat4 proj = glm::mat4(1.0f);
+    };
+
+    ReginaContext g_regina_ctx; // NOLINT(fuchsia-statically-constructed-objects, cppcoreguidelines-avoid-non-const-global-variables)
+
+    //-------------------------------------------------------------------------
+    renderer::ResourceSlot get_active_object_commited_resource_for_frame(s32 frame)
```

Comment:
typo here, committed is with double t

Reply:
I think you want to pass in the ResourceSlot here, not an int. anyone can call this with a random int, which is not really what you want. I think this is part of a design problem of ResourceSlot which I highlighted in its header

Reply:
I'd also rename it to "get_active_heap_for_object_cbs" or something

Reply:
Don't know why this is a design flaw I can just change it to ResourceSlot and be done with it. Think I just forgot to change these.

I renamed it to get_commited_resource_of_frame(/*frame*/)